### PR TITLE
Hybrid model for ProtoDUNE-HD

### PR DESCRIPTION
**Pull request by Li-Jiaoyang97**
Added the parameterisations for semi-analytical model and photonvisibilityservice for photon library part.

### CODE REVIEW

File: duneopdet/PhotonPropagation/photpropservices_dune.fcl [hunk 1]
```diff
@@ -418,6 +414,29 @@ protodunedp_driftY_AV_photonvisibilityservice:
...
+  # hybrid library
+  NX: 122
+  NY: 67
+  NZ: 93
+
+  UseCryoBoundary: false
+  DoNotLoadLibrary: false
+  LibraryBuildJob: false
+
+  LibraryFile: "PhotonPropagation/LibraryData/Photon_library_protoDUNEhd_library_protoDUNEhd_v2_refactored_nonActive.root"
```
Comment (pgreen135) [inline duneopdet/PhotonPropagation/photpropservices_dune.fcl:431]: is this name correct? "...library_protoDUNEhd_library_protoDUNEhd..."

**Review by pgreen135 — APPROVED** [review]
Otherwise looks good, thanks @Li-Jiaoyang97

File: duneopdet/PhotonPropagation/photpropservices_dune.fcl [hunk 2]
```diff
@@ -418,6 +414,29 @@ protodunedp_driftY_AV_photonvisibilityservice:
...
+{
+  # hybrid library
+  NX: 122
+  NY: 67
+  NZ: 93
+
+  UseCryoBoundary: false
+  DoNotLoadLibrary: false
+  LibraryBuildJob: false
+
```
Comment (lpaulucc) [inline duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: In the hybrid model, is the option IncludePropTime available?

**Comment by lpaulucc** [comment]
@Li-Jiaoyang97 I left a comment in photpropservices_dune.fcl

File: duneopdet/PhotonPropagation/photpropservices_dune.fcl (hunk 1 above)
Comment (Li-Jiaoyang97) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:431]: Unfortunately, it's the correct name. (This library will be regenerated again when the field cage is included in the geometry. I will correct the name for the new library. )

File: duneopdet/PhotonPropagation/photpropservices_dune.fcl (hunk 2 above)
Comment (Li-Jiaoyang97) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: Hi Laura. I haven't put them into the hybrid model, can anyone please remind me what IncludePropTime is supposed to be used here? Thanks.

Comment (lpaulucc) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: I think it should be the photon propagation time, no? Maybe not relevant here... I am just checking.

Comment (pgreen135) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: It should be turned on, with for now the same parameters as inside the volume. Not an ideal solution (a dedicated model will probably be desired eventually), but without it the photons from outside the active volume will incorrectly arrive earlier than those from inside. This then can cause various problems when calculating t0 etc. down the line (we had issues with this in SBND).

Comment (lpaulucc) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: Thanks, Patrick. Can you make the change, Jiaoyang?

Comment (Li-Jiaoyang97) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:430]: Thanks, Laura. Thanks, Patrick. I will make the change for both ProtoDUNE-HD and DUNE-VD geometry.

Comment (Li-Jiaoyang97) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:431]: Hi all, I have made the changes.

Comment (lpaulucc) [reply duneopdet/PhotonPropagation/photpropservices_dune.fcl:431]: Great, Jiaoyang. Thanks!

**Review by lpaulucc — APPROVED** [review]

**Pull request merged**